index.md

```diff
@@ -2,9 +2,286 @@
+Staff will still be able to access it from the USDF RSP using a direct butler connection.
+The dataset can be used as a client/server testbed for the hybrid data access center approach where the client is on Google but the data are at SLAC.
+
+### ComCam Data
+
+* ComCam data (both raw and reduced) will not be made available to the Data Rights community until Data Preview 1 (DP1) {cite:p}`RTN-011` in early 2025.
+  This includes access to prompt processing products.
+* DP1 will use Butler client/server.
+* The initial plan is for DP1 to be read-only with no ability to run `Butler.put()`.
+* Since this is a static read-only dataset the ObsCore tables can be populated by exporting from the Butler registry and importing into Qserv {cite:p}`DMTN-243` as was done for DP0.2.
+* Since this data repository is read-only there is no need for group or user collection management.
+
+### Prompt Processing During Survey Operations
+
+As soon as we start taking survey data the Data Rights community will expect to be able to retrieve the raw data and prompt processing outputs (once embargo period for that night has elapsed).
+This will use a Butler client/server, likely with the registry and data being located at SLAC.
+To support VO services "live" ObsCore will be used with Butler updating the ObsCore tables as needed {cite:p}`DMTN-236`.
+
+## Implementation Decisions
+
+### Querying
+
+With potentially 10,000 simultaneous users running queries for datasets and dimension records, it is impossible for one server to be able to handle that load with one database.
+To solve this problem there will be multiple strategies:
+
+* Servers will cache some database dimension records.
+* Multiple server instances will be deployed with load balancing.
+* Multiple database server instances will be deployed.
+* Servers will not run database queries directly, instead they will spawn workers that will run the queries.
+
+
+#### Record Caching
+
+For a static data release the dimension coordinate space is pre-defined and no new records will be added.
+Even for a live data repository, such as the Prompt Processing repository, certain record categories are only rarely augmented and are relatively small.
+
+For example, the number of `instrument` records is of order 10, the number of `physical_filter` records is of order 1,000, and the number of `detector` records is of order 1,000.
+These are very common dimensions and caching them in the server is very important to reduce database load.
+If a new filter is added, this event can be planned for and a mechanism can be in place to invalidate the cache to allow it to be recreated.
+
+Even for `day_obs` the total number of records is only about 4,000 for LSSTCam over the ten year survey, and for a static data release we could consider caching those.
+
+The situation is different for `exposure`, `visit` and `group` where there can be of order 1,000 of each of those created per night.
+Those records can not be cached.
```

Comment:
Maybe we should stop calling this "caching", because what we really do with dimension records is just "aggressive fetching" - we just pull them all to the client up front.

Reply:
For a static release exposure+visit+group won't change but in DR11 there will be many records. For prompt processing each day there will be more of them. I imagine we could have a flag in a butler config to say whether it's thought of as a static set of dimensions or dynamic set.

Don't we have to cache in the server even if we are caching in the clients otherwise 100 butler object instantiations hitting the server all at once will cause the server to hit registry 100 times.